driver/src/contracts/stablex_contract.rs

```diff
@@ -139,40 +141,38 @@ pub trait StableXContract {
 }
 
 impl StableXContract for StableXContractImpl {
-    fn get_current_auction_index(&self) -> Result<u32> {
-        let auction_index = self.instance.get_current_batch_id().call().wait()?;
-        Ok(auction_index)
+    fn get_current_auction_index(&self) -> BoxFuture<'static, Result<u32>> {
+        let future = self.instance.get_current_batch_id().call();
+        async move { future.await.map_err(Error::from) }.boxed()
```

Comment:
Is there a reason you prefer this over [`TryFutureExt::map_err`](https://docs.rs/futures/0.3.5/futures/future/trait.TryFutureExt.html#method.map_err)?

Reply:
No. Just didn't know about that.

Reply:
Seeing as I want to have an async block anyway to make it look like this is an async function I am keeping it like this.